src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -63,24 +63,18 @@ internal static string EnsureTrailingDirectorySeparator(string fullPath)
         //
         public static bool Exists(string path)
         {
-            try
+            if (string.IsNullOrEmpty(path) || path.Contains('\0'))
+                return false;
+            
+            if (!path.StartsWith(@"\\?\") && !path.StartsWith(@"\??\"))
             {
-                if (path == null)
-                    return false;
-                if (path.Length == 0)
-                    return false;
-
-                string fullPath = Path.GetFullPath(path);
-
-                return FileSystem.DirectoryExists(fullPath);
+				if ( path.Contains('\\') && path.Substring(path.LastIndexOf('\\')) == "\\..")
+				{
+					path = Path.GetDirectoryName(Path.GetDirectoryName(path));
+				}
+                path = path.TrimEnd(new char[] { '.', ' ' });
```

Comment:
Wherever this logic ends up living, we shouldn't be allocating a new array like this each time.